### PR TITLE
Add pyteal example

### PR DESCRIPTION
**Pull request by amityadav0**

### CODE REVIEW

**Review by ratik21 — COMMENTED** [review]
@amityadav0 please add a readme explaining the example and setup process. Also add some links as well.

File: examples/pyteal/algob.config.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    mnemonic: "enforce drive foster uniform cradle tired win arrow wasp melt cattle chronic sport dinosaur announce shell correct shed amused dismiss mother jazz task above hospital"
+  },
+  // Following accounts are generated using `algob gen-accounts`.
+  // Do not use these exact accounts in your own applications.
+  // See notice at the top of this file.
+  {
+    name: "gold-owner-account",
+    addr: "M7VR2MGHI35EG2NMYOF3X337636PIOFVSP2HNIFUKAG7WW6BDWDCA3E2DA",
+    mnemonic: "quick stage planet wild coffee whale build brisk forest leave same segment armed alter fog know run distance excess galaxy limb talent nut able noodle"
+  },
```
Comment (ratik21) [inline examples/pyteal/algob.config.js:28]: are we using `gold-owner` account?

File: examples/pyteal/scripts/transfer/hltc-example.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/**
+ * Description:
+ * This file demonstrates the PyTeal Example for HLTC(Hash Time Lock Contract)
+*/
```
Comment (ratik21) [inline examples/pyteal/scripts/transfer/hltc-example.js:4]: Please add one more line explaining about HLTC.

Comment (amityadav0) [reply examples/pyteal/scripts/transfer/hltc-example.js:8]: ok

File: examples/pyteal/algob.config.js (hunk 1 above)
Comment (amityadav0) [reply examples/pyteal/algob.config.js:28]: no, this will be removed

**Comment by amityadav0** [comment]
@ratik21 this is simpler example than multisig (similar to deployment examples), i have added few comments on top of function description.
Please let me know if it gives clear picture. If not i will add a readme.md with more comments.

**Comment by ratik21** [comment]
@amityadav0 yes this is simpler. But since this is a new project we should have a readme. You can add a basic one.

File: examples/pyteal/scripts/py-teal.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+async function run(runtimeEnv, deployer) {
+
+  const masterAccount = deployer.accountsByName.get("master-account")
+  const bobAccount = deployer.accountsByName.get("bob-account"); // Buyer
+
+  await transferMicroAlgos(deployer, masterAccount, bobAccount.addr, 200000000, {note: "funding account"});  
+
+  // Secret value // Image : QzYhq9JlYbn2QdOMrhyxVlNtNjeyvyJc/I8d8VAGfGc=
+  const secret = "hero wisdom green split loop element vote belt";
+  const hltcInfoAlgoContract = await deployer.fundLsig("hltc.py", [ secret ],
```
Comment (robert-zaremba) [inline examples/pyteal/scripts/py-teal.js:16]: this should be renamed to `htlc...` (note reverse between t and l)

File: examples/pyteal/scripts/py-teal.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+const { transferMicroAlgos, transferAsset, balanceOf, ASC1Mode } = require("algob");
+
+async function run(runtimeEnv, deployer) {
+
+  const masterAccount = deployer.accountsByName.get("master-account")
+  const bobAccount = deployer.accountsByName.get("bob-account"); // Buyer
+
+  await transferMicroAlgos(deployer, masterAccount, bobAccount.addr, 200000000, {note: "funding account"});  
+
+  // Secret value // Image : QzYhq9JlYbn2QdOMrhyxVlNtNjeyvyJc/I8d8VAGfGc=
```
Comment (robert-zaremba) [inline examples/pyteal/scripts/py-teal.js:14]: is the `Qz...` a base64 encoding?

File: examples/pyteal/scripts/py-teal.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+/**
+ * Description:
+ * This file demonstrates the PyTeal Example for HLTC(Hash Time Lock Contract)
```
Comment (robert-zaremba) [inline examples/pyteal/scripts/py-teal.js:3]:
```suggestion
 * This file demonstrates the PyTeal Example for HTLC(Hash Time Lock Contract)
```

File: examples/pyteal/scripts/transfer/hltc-example.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    const sender = lsig.address(); 
+    
+    // Transaction Passes : as right secret value is used
+    await transferAlgo(deployer, { addr: sender}, johnAccount.addr, 200, lsig);
+
+    const lsig1 = await deployer.loadLsig("hltc.py", [ wrongSecret ]);
+    const sender1 = lsig1.address();
+
+    // Transaction Fails : as wrong secret value is used
+    await transferAlgo(deployer, { addr: sender1}, johnAccount.addr, 200, lsig1);
```
Comment (robert-zaremba) [inline examples/pyteal/scripts/transfer/hltc-example.js:29]: please change the order, firstly the wrong one then the good one.

You don't need to redeclare lsig1, sender1 - you can use `let` and reuse lsig and sender.

File: examples/pyteal/assets/hltc.py [hunk 7]
```diff
@@ -0,0 +1,36 @@
+# Hash Time Lock Contract Example in pyTeal
```
Comment (robert-zaremba) [inline examples/pyteal/assets/hltc.py:1]: please rename this file to `htlc.py`

**Review by robert-zaremba — CHANGES_REQUESTED** [review]
Agree with Ratik - we should add README with instructions how to run it (algob run ..)

File: examples/pyteal/scripts/transfer/hltc-example.js (hunk 6 above)
Comment (amityadav0) [reply examples/pyteal/scripts/transfer/hltc-example.js:29]: ok

File: examples/pyteal/assets/hltc.py (hunk 7 above)
Comment (amityadav0) [reply examples/pyteal/assets/hltc.py:1]: ok

File: examples/pyteal/scripts/py-teal.js (hunk 3 above)
Comment (amityadav0) [reply examples/pyteal/scripts/py-teal.js]: ok

File: examples/pyteal/scripts/py-teal.js (hunk 4 above)
Comment (amityadav0) [reply examples/pyteal/scripts/py-teal.js]: secret value hashed with sha256 will produce our image hash. which will be used in Contract code for later matching

Comment (amityadav0) [reply examples/pyteal/scripts/py-teal.js]: adding a more elaborated comment

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**